apps/prairielearn/src/lib/score-upload.js

```diff
@@ -9,372 +9,387 @@ const manualGrading = require('./manualGrading');
+    try {
+      feedback_obj = JSON.parse(feedback_json);
+    } catch (e) {
+      throw new Error(`Unable to parse "feedback_json" field as JSON: ${e}`);
+    }
+    if (!_.isPlainObject(feedback_obj)) {
+      throw new Error(`Parsed "feedback_json" is not a JSON object: ${feedback_obj}`);
+    }
+    feedback = feedback_obj;
     if (feedback_string != null) {
-      feedback = { manual: feedback_string };
+      feedback.manual = feedback_string;
     }
-    if (feedback_json != null) {
-      let feedback_obj = null;
-      try {
-        feedback_obj = JSON.parse(feedback_json);
-      } catch (e) {
-        throw new Error(`Unable to parse "feedback_json" field as JSON: ${e}`);
-      }
-      if (!_.isPlainObject(feedback_obj)) {
-        throw new Error(`Parsed "feedback_json" is not a JSON object: ${feedback_obj}`);
-      }
-      feedback = feedback_obj;
-      if (feedback_string != null) {
-        feedback.manual = feedback_string;
-      }
+  }
+  return feedback;
+}
+
+function _getPartialScoresOrNull(json) {
+  const partial_scores_json = _getJsonPropertyOrNull(json, 'partial_scores');
+  let partial_scores = null;
+  if (partial_scores_json != null) {
+    try {
+      partial_scores = JSON.parse(partial_scores_json);
+    } catch (e) {
+      throw new Error(`Unable to parse "partial_scores" field as JSON: ${e}`);
     }
-    return feedback;
-  },
-
-  _getPartialScoresOrNull(json) {
-    const partial_scores_json = module.exports._getJsonPropertyOrNull(json, 'partial_scores');
-    let partial_scores = null;
-    if (partial_scores_json != null) {
-      try {
-        partial_scores = JSON.parse(partial_scores_json);
-      } catch (e) {
-        throw new Error(`Unable to parse "partial_scores" field as JSON: ${e}`);
-      }
-      if (!_.isPlainObject(partial_scores)) {
-        throw new Error(`Parsed "partial_scores" is not a JSON object: ${partial_scores}`);
-      }
+    if (!_.isPlainObject(partial_scores)) {
+      throw new Error(`Parsed "partial_scores" is not a JSON object: ${partial_scores}`);
     }
-    return partial_scores;
-  },
-
-  async _updateInstanceQuestionFromJson(json, assessment_id, authn_user_id) {
-    const submission_id = module.exports._getJsonPropertyOrNull(json, 'submission_id');
-    const uid_or_group =
-      module.exports._getJsonPropertyOrNull(json, 'group_name') ??
-      module.exports._getJsonPropertyOrNull(json, 'uid');
-    const ai_number = module.exports._getJsonPropertyOrNull(json, 'instance');
-    const qid = module.exports._getJsonPropertyOrNull(json, 'qid');
-
-    await sqldb.runInTransactionAsync(async () => {
-      const submission_data = await sqldb.queryZeroOrOneRowAsync(sql.select_submission_to_update, {
-        assessment_id,
-        submission_id,
-        uid_or_group,
-        ai_number,
-        qid,
-      });
+  }
+  return partial_scores;
+}
+
+async function _updateInstanceQuestionFromJson(json, assessment_id, authn_user_id) {
```

Comment:
Nit: I'd appreciate a JSDOC comment explaining the return value.

Reply:
Done in 90a750863951ea070af2a5ee64d914df40cac80d.